SSTTP-1184 - Calculator input will be reset upon sign in

cavster opened the pull request:
https://jira.tools.tax.service.gov.uk/browse/SSTTP-1184

cavster committed:
Calculator input will be reset upon sign in

nkaraolis commented on app/uk/gov/hmrc/selfservicetimetopay/controllers/EligibilityController.scala:
```diff
@@ -69,13 +69,15 @@ class EligibilityController @Inject() (val messagesApi: play.api.i18n.MessagesAp
   def submitSignInQuestion: Action[AnyContent] = Action.async { implicit request =>
     sessionCache.get.map {
-      case Some(TTPSubmission(_, _, _, _, `validTypeOfTax`, `validExistingTTP`, cd, _, _, _)) =>
+      case Some(tpp:TTPSubmission)if tpp.eligibilityExistingTtp == `validExistingTTP` && tpp.eligibilityTypeOfTax == `validTypeOfTax`  =>
```
Rename to ttp to retain consistency throughout project

nkaraolis commented on app/uk/gov/hmrc/selfservicetimetopay/controllers/EligibilityController.scala:
```diff
@@ -69,13 +69,15 @@ class EligibilityController @Inject() (val messagesApi: play.api.i18n.MessagesAp
           {
-            case SignInQuestion(Some(true)) => Redirect(routes.ArrangementController.determineMisalignment())
+            case SignInQuestion(Some(true)) =>
+              if(tpp.calculatorData.debits.nonEmpty) sessionCache.put(tpp.copy(calculatorData = CalculatorInput.initial))
```
Put method returns a future so might not be completed in time before redirect so probably best to flatMap on the initial future and then map on this one

cavster replied:
Hmmm take a look at submitAmountOwed in the Calculator controller then ;)

nkaraolis replied:
Good spot, might as well update that in this PR as well

cavster replied:
I'll let you have the fun of doing it ;)

cavster committed:
reworth the seesion put to use flat map instead to avoid race conditions

nkaraolis committed:
Updated sessionCache put in submitAmountOwed to make sure future is completed before redirect

Commit:
Changed format of methods and fixed broken test

calculator controller test was broken as a sessionCache.put was
not being mocked. This didn't exist before due to a race condition,
where the put was not being waited for before a redirect occured.
